src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/DataSetQueryService.cs

```diff
@@ -133,7 +133,11 @@ private async Task<Either<ActionResult, DataSetVersion>> FindDataSetVersion(
             .DataSetVersions
             .AsNoTracking()
             .Include(dsv => dsv.DataSet)
-            .FindByVersion(dataSetId, dataSetVersion, cancellationToken);
+            .FindByVersion(
+                dataSetId: dataSetId, 
+                version: dataSetVersion,
+                publicOnly: true, 
+                cancellationToken);
     }
```

Comment:
Just wondering if above on L120 where we already had `if (dataSetVersion is null)`, if this should be altered to cover null **OR** '*'?

That would mean if the version is a wildcard then it would fall into the block above this one which uses `.Select(ds => ds.LatestLiveVersion!)` which could be an optimisation over retrieving the top public version from all the versions after sorting.